src/python/code/gnoll/__main__.py

```diff
@@ -0,0 +1,48 @@
+'''Roll some dice with GNOLL.'''
+
+import sys, argparse, gnoll
+
+def parse_cmdline_args(args):
+    p = argparse.ArgumentParser(
+        description = __doc__,
+        usage = 'python3 -m gnoll [options] EXPR',
+        add_help = False)
+
+    p.add_argument('EXPR', nargs = '+',
+        help = 'a dice expression to evaluate '
+            '(multiple arguments will be joined with spaces)')
+
+    g = p.add_argument_group('main options')
+    g.add_argument('-h', '--help', action = 'help',
+        help = 'show this help message and exit')
+    g.add_argument('-b', '--breakdown', action = 'store_true',
+        help = 'show a breakdown into individual dice')
+    g.add_argument('--no-builtins', action = 'store_true',
+        help = 'disable built-in macros')
+
+    g = p.add_argument_group('debugging options')
+    g.add_argument('-v', '--verbose', action = 'store_true',
+        help = 'enable verbosity')
+    g.add_argument('--keep-temp-file', action = 'store_true',
+        help = "don't delete the created temporary file")
+    g.add_argument('--mock', metavar = 'TYPE', type = int,
+        help = 'mocking type')
+    g.add_argument('--mock-const', metavar = 'N', type = int, default = 3,
+        help = 'mocking constant')
+
+    a = p.parse_args(args)

```

Comment:
**suggestion (non-blocking):** You may be able to call this without the args object - I think argparse will automatically collect them for you - then you would be able to remove the parsing code

```suggestion
    a = p.parse_args()
```

Reply:
That's true in the usual case, but I figured that allowing `parse_cmdline_args` to be called with an argument would be convenient for testing. Then you don't have to mock `sys.argv`.